Infrastructure/Security/IsHostRequirement.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.EntityFrameworkCore;
+using Persistence.Data;
+using System.Security.Claims;
+
+namespace Infrastructure.Security
+{
+    public class IsHostRequirement : IAuthorizationRequirement
+    {
+    }
+    public class IsHostRequirementHandler : AuthorizationHandler<IsHostRequirement>
+    {
+        private readonly DataContext _dataContext;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public IsHostRequirementHandler(DataContext dataContext, IHttpContextAccessor httpContextAccessor)
+        {
+            _dataContext = dataContext;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, IsHostRequirement requirement)
+        {
+            if (context.Resource is AuthorizationFilterContext authContext)
+            {
+                var currentUserName = _httpContextAccessor.HttpContext.User?.Claims?.SingleOrDefault(x => x.Type == ClaimTypes.NameIdentifier)?.Value;
+
+                var activityId = Guid.Parse(authContext.RouteData.Values["id"].ToString());
+
+                var activity = _dataContext.Activities.FindAsync(activityId).Result;
+
+                var host = activity.Attendees.FirstOrDefault(x => x.IsHost);
+
+                if (host?.AppUser?.UserName == currentUserName)
+                    context.Succeed(requirement);
+            }
+            else
+            {
```

Comment:
reverse the if check and treat the sad path first. So that the rest of the method is not enclosed in an if statement.

```
if() 
{ 
context.Fail();
return Task.CompletedTask;
}

[rest of the logic]
return Task.CompletedTask; 
```